js/rook.js

```diff
@@ -0,0 +1,86 @@
+import {Piece} from "./piece.js";
+
+export class Rook extends Piece {
+    /**
+     * Constructor: inherits from abstract Piece class
+     * @param id piece id
+     * @param rowPos current row position
+     * @param colPos current column position
+     * @param playerId player id
+     */
+    constructor(id, rowPos, colPos, playerId) {
+        super();
+        this.id = id;
+        this.rowPos = rowPos;
+        this.colPos = colPos;
+        this.playerId = playerId;
+        this.unicodeChar = "&#9814;";
+    }
+
+    /**
+     * Moves piece to {row}, {col} and updates board
+     * @param row new row
+     * @param col new column
+     * @param board Board.board
+     */
+    movePiece(row, col, board) {
+        let possibleMoves = this.possibleMoves(board);
+        if (!this.validateMove(row, col, possibleMoves)) { // ignore invalid move
+            return;
+        }
+
+        if (board[row][col]) { // taking a piece
+            this.takenPiece = board[row][col];
+            this.takenPiece.setTaken();
+            board[row][col] = null;
+        }
```

Comment:
Don't forget to handle the case where a piece isn't being taken; the piece's `takenPiece` attribute should be set to `null` in that case.

Reply:
Added, thank you!